specs/waku/v2/waku-v2-rpc-api.md

```diff
@@ -0,0 +1,459 @@
+---
+title: Waku v2 JSON-RPC REST API
+version: 0.0.1
+status: Raw
+authors: Hanno Cornelius <hanno@status.im>
+---
+
+# Table of Contents
+
+1. [Introduction](#introduction)
+1. [Wire Protocol](#wire-protocol)
+    1. [Transport](#transport)
+    1. [Types](#types)
+    1. [Method naming](#method-naming)
+    1. [Debug API](#debug-api)
+    1. [Relay API](#relay-api)
+    1. [Store API](#store-api)
+    1. [Filter API](#filter-api)
+1. [Example usage](#example-usage)
+1. [Changelog](#changelog)
+1. [Copyright](#copyright)
+
+# Introduction
+
+This specification describes the JSON-RPC API that Waku v2 nodes MAY adhere to. Refer to the [Waku v2 specification](https://github.com/vacp2p/specs/blob/master/specs/waku/v2/waku-v2.md) for more information on Waku v2.
+
+# Wire Protocol
+
+## Transport
+
+Nodes SHOULD expose an accessible [JSON-RPC](https://www.jsonrpc.org/specification) API. The JSON-RPC version SHOULD be `2.0`. Below is an example request:
+
+```json
+{
+  "jsonrpc":"2.0",
+  "method":"get_waku_v2_debug_info",
+  "params":[],
+  "id":1
+}
+```
+
+### Fields
+
+| Field     | Description                                         |
+| --------- | --------------------------------------------------- |
+| `jsonrpc` | Contains the used JSON-RPC version (`Default: 2.0`) |
+| `method`  | Contains the JSON-RPC method that is being called   |
+| `params`  | An array of parameters for the request              |
+| `id`      | The request ID                                      |
+
+## Types
+
+In this specification, the primitive types `Boolean`, `String`, `Number` and `Null`, as well as the structured types `Array` and `Object`, are to be interpreted according to the [JSON-RPC specification](https://www.jsonrpc.org/specification#conventions). It also adopts the same capitalisation conventions.
+
+The following structured types are defined for use throughout the document:
+
+### WakuMessage
+
+Refer to [`Waku Message` specification](https://github.com/vacp2p/specs/blob/master/specs/waku/v2/waku-message.md) for more information.
+
+`WakuMessage` is an `Object` containing the following fields:
+| Field | Type | Inclusion | Description |
+| ----: | :--: | :--: |----------- |
+| `payload` | `String` | mandatory | The message payload |
+| `contentTopic` | `Number` | optional | Message content topic for optional content-based filtering |
```

Comment:
I think this has been talked about, and I know contentTopic is `fixed32` in WakuMessage (for bridging with v1 purposes), but did we consider this being something more user input friendly like a hex string? E.g. in v1 this is a string (https://github.com/vacp2p/specs/blob/master/specs/waku/v1/waku-rpc-api.md, note that topic here corresponds to content topic in v2).

Reply:
You're right. In general a try to avoid using `Number` as user input unless you're actually counting something (like time or messages or pages). Conversion from hex string should be easy to add.